posts/2023-09-29-open-liberty-on-aws-ecs-fargate.adoc

```diff
@@ -0,0 +1,684 @@
+            "cpu": 512,
+            "memory": 1024,
+            "portMappings": [
+                {
+                    "name": "liberty-getting-started-9080-tcp",
+                    "containerPort": 9080,
+                    "hostPort": 9080,
+                    "protocol": "tcp",
+                    "appProtocol": "http"
+                },
+                {
+                    "name": "liberty-getting-started-9443-tcp",
+                    "containerPort": 9443,
+                    "hostPort": 9443,
+                    "protocol": "tcp",
+                    "appProtocol": "http"
+                }
+            ],
+            "essential": true,
+            "environment": [],
+            "environmentFiles": [],
+            "mountPoints": [],
+            "volumesFrom": [],
+            "logConfiguration": {
+                "logDriver": "awslogs",
+                "options": {
+                    "awslogs-create-group": "true",
+                    "awslogs-group": "/ecs/ol-getting-started-demo",
+                    "awslogs-region": "us-east-1",
+                    "awslogs-stream-prefix": "ecs"
+                }
+            }
+        }
+    ],
+    "executionRoleArn": "",
+    "networkMode": "awsvpc",
+    "requiresCompatibilities": [
+        "FARGATE"
+    ],
+    "cpu": "512",
+    "memory": "1024",
+    "runtimePlatform": {
+        "cpuArchitecture": "X86_64",
+        "operatingSystemFamily": "LINUX"
+    }
+}
+----
+
+A full list of task definition parameters can be found https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definition_parameters.html[here].
+
+=== Configuring the Network
+While we are using Amazon ECS with AWS Fargate to manage our runtime, networking components come under Amazon EC2 and still require an amount of management.
+
+For this blog, we want to expose our application externally on protocol default ports: `80` for HTTP and `443` for HTTPS, while using non-default protocol ports for our container.
+
+The networking components can be created during the Amazon ECS Service creation steps. However, issues have been encountered when creating everything using the Amazon ECS Service creation wizard. The primary issue is mapping the different port values between the task definition and the exposing load balancer. The wizard contains an assumption that the port that we will publically expose the service by, is the same as the port that the target group will be forwarding to on the running container. As the port mapping cannot be changed once the target group has been created, the networking components will be created first and then referenced by the service later.
+
+For this blog the default Virtual Private Cloud(VPC) is sufficient and for the standard configuration we will use `HTTP` as the chosen protocol. For HTTPS, <<_securing_your_service, Securing your service>> goes through the additional steps to secure your service with certificates.
```

Comment:
These instructions seem really long and make this task look quite complex and difficult. Could any of this be cut down and/or put into a separate blog? Perhaps the “securing the service” section could be a separate post? Or, even the manual and auto-scaling sections (unless it would be weird to not include them in the basic post).